### DISCUSSION TITLE
What is the best practise with .terraform.lock.hcl in modules … commit or not?

### DISCUSSION THREAD

**Discussion by yasinlachiny** [particuleio/teks#155] [Q&A]
Is it not better to keep `.terraform.lock.hcl` in our repo?
For example, it could be more secure

**Comment by acutchin-bitpusher** [comment]
Essential to keep in the repo. Until you deliberately decide to upgrade providers/versions, you want to use the same exact set in Terraform actions, regardless of where/how/when they run and who or what runs them.

**Reply by yasinlachiny** [reply]
What if we hardcode the version in the terraform-version file?

**Reply by acutchin-bitpusher** [reply]
You mean the version of every provider? I suppose that is another way to do it.
